snuba/web/query.py

```diff
@@ -2,263 +2,35 @@
-                            }
-                        )
-
-                        if use_cache:
-                            cache.set(query_id, result)
-                            timer.mark("cache_set")
-
-                    except BaseException as ex:
-                        error = str(ex)
-                        logger.exception("Error running query: %s\n%s", sql, error)
-                        stats = update_with_status("error")
-                        meta = {}
-                        if isinstance(ex, ClickhouseError):
-                            err_type = "clickhouse"
-                            meta["code"] = ex.code
-                        else:
-                            err_type = "unknown"
-                        raise RawQueryException(
-                            err_type=err_type,
-                            message=error,
-                            stats=stats,
-                            sql=sql,
-                            **meta,
-                        )
-            except RateLimitExceeded as ex:
-                stats = update_with_status("rate-limited")
-                raise RawQueryException(
-                    err_type="rate-limited",
-                    message="rate limit exceeded",
-                    stats=stats,
-                    sql=sql,
-                    detail=str(ex),
-                )
-
-    stats = update_with_status("success")
-
-    return RawQueryResult(result, {"stats": stats, "sql": sql})
-
-
-def update_query_metadata_and_stats(
-    request: Request,
-    sql: str,
-    timer: Timer,
-    stats: MutableMapping[str, Any],
-    query_metadata: SnubaQueryMetadata,
-    query_settings: Mapping[str, Any],
-    trace_id: Optional[str],
-    status: str,
-) -> MutableMapping:
-    """
-    If query logging is enabled then logs details about the query and its status, as
-    well as timing information.
-    Also updates stats with any relevant information and returns the updated dict.
-    """
-
-    stats.update(query_settings)
-
-    query_metadata.query_list.append(
-        ClickhouseQueryMetadata(sql=sql, stats=stats, status=status, trace_id=trace_id)
-    )
-
-    return stats
-
-
-def record_query(
-    request: Request, timer: Timer, query_metadata: SnubaQueryMetadata
-) -> None:
-    if settings.RECORD_QUERIES:
-        # Send to redis
-        # We convert this to a dict before passing it to state in order to avoid a
-        # circular dependency, where state would depend on the higher level
-        # QueryMetadata class
-        state.record_query(query_metadata.to_dict())
-
-        final = str(request.query.get_final())
-        referrer = request.referrer or "none"
-        timer.send_metrics_to(
-            metrics,
-            tags={
-                "status": query_metadata.status,
-                "referrer": referrer,
-                "final": final,
-            },
-            mark_tags={"final": final},
-        )
-
-
 def parse_and_run_query(
```

Comment:
This is the entry point now. Did not touch that not to break everything else in the codebase. 
But, after moving split query in the Query Execution Strategy for Storage split, this method will not need to exist anymore.